plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterActionWorker.java

```diff
@@ -361,7 +361,7 @@ protected IpAddress getVpcTierKubernetesPublicIp(Network network) {
             return null;
         }
         IpAddress address = ipAddressDao.findByUuid(detailsVO.getValue());
-        if (address == null || network.getVpcId() != address.getVpcId()) {
+        if (address == null || network.getVpcId() == null || address.getVpcId() == null || network.getVpcId().longValue() != address.getVpcId().longValue()) {
```

Comment:
```suggestion
        if (address == null || !network.getVpcId().equals(address.getVpcId())) {
```

Before calling this method, we're already verifying that `network.getVpcId()` is not null

Reply:
possibly so @winterhazel , but not in this protected method and we have no contract on it, so we still need to check inside the method.

that said
```suggestion
        if (address == null || !address.getVpcId().equals(network.getVpcId()) {
```
makes for an easier contract (network not null  instead of some deep DB involved relation)

As we have no control now on how this may be called I would leave some kind of null check in at least.